ui/src/views/Clients/ViewClients/index.jsx

```diff
@@ -81,6 +64,13 @@ export default class ViewClients extends PureComponent {
         limit: VIEW_CLIENTS_PAGE_SIZE,
       },
     });
+
+    // Prevent searching for the same query as the one already loaded
+    if (clientSearch !== this.props.history.location.search.substring(8)) {
+      this.props.history.push(
+        clientSearch.length > 0 ? `?search=${clientSearch}` : '/auth/clients'
```

Comment:
Let's have the client search term encoded via `encodeURIComponent` before pushing it to the url. To decode, you can use `decodeURIComponent`.